### DISCUSSION TITLE
How do I use MSA transformer without the contact head

### DISCUSSION THREAD

**Discussion by DanBuchan** [facebookresearch/esm#507] [Q&A]
I note that on loading the msa-transformer model/weights that the layers and weights for both for `contact_head.regression.weight` and `contact_head.regression.bias` are present if I list the layers. Do I need to remove these layers from the model to use it without the contact head or is it sufficient to just set `return_contacts=False` when running a forward pass of an instance of MSATransformer?

**Comment by tomsercu** [comment]
These weights, and the computation, are super minimal so they shouldn't be in the way from a GPU memory perspective. Just `return_contacts=False` should be enough for 99.9% of usecases

**Comment by DanBuchan** [comment]
Great, thanks
